code/modules/power/breaker_box.dm

```diff
@@ -0,0 +1,104 @@
+	var/icon_state_on = "bbox_on"
+	var/icon_state_off = "bbox_off"
+	density = 1
+	anchored = 1
+	var/on = 1
+	var/busy = 0
+	var/directions = list(1,2,4,8,5,6,9,10)
+	var/update_locked = 0
+
+/obj/machinery/power/breakerbox/Initialize()
+	. = ..()
+	set_state(1)
+
+/obj/machinery/power/breakerbox/examine(mob/user)
+	. = ..()
+	to_chat(user, "Large machine with heavy duty switching circuits used for advanced grid control")
+	if(on)
+		to_chat(user, "<span class='notice'>It seems to be online.</span>")
+	else
+		to_chat(user, "<span class='warning'>It seems to be offline.</span>")
+
+/obj/machinery/power/breakerbox/attack_ai(mob/user)
+	if(update_locked)
+		to_chat(user, "<span class='warning'>System locked. Please try again later.</span>")
+		return
+
+	if(busy)
+		to_chat(user, "<span class='warning'>System is busy. Please wait until current operation is finished before changing power settings.</span>")
+		return
+
+	busy = 1
+	to_chat(user, "<span class='good'>Updating power settings..</span>")
+	if(do_after(user, 50, src))
+		set_state(!on)
+		to_chat(user, "<span class='good'>Update Completed. New setting:[on ? "on": "off"]</span>")
+		update_locked = 1
+		spawn(50) //AI isn't slow
+			update_locked = 0
+	busy = 0
+
+/obj/machinery/power/breakerbox/attack_hand(mob/user)
+	if(update_locked)
+		to_chat(user, "<span class='warning'>System locked. Please try again later.</span>")
+		return
+
+	if(busy)
+		to_chat(user, "<span class='warning'>System is busy. Please wait until current operation is finished before changing power settings.</span>")
+		return
+
+	busy = 1
+	for(var/mob/O in viewers(user))
+		O.show_message(text("<span class='warning'>\The [user] starts reprogramming \the [src]!</span>"), 1)
+
+	if(do_after(user, 50,src))
+		set_state(!on)
+		user.visible_message(\
+		"<span class='notice'>[user.name] [on ? "enabled" : "disabled"] the breaker box!</span>",\
+		"<span class='notice'>You [on ? "enabled" : "disabled"] the breaker box!</span>")
+		update_locked = 1
+		spawn(300) //Humans are slow
+			update_locked = 0
+	busy = 0
+
+/obj/machinery/power/breakerbox/proc/set_state(var/state)
+	on = state
+	if(on)
+		icon_state = icon_state_on
+		var/list/connection_dirs = list()
+		for(var/direction in directions)
+			for(var/obj/structure/cable/C in get_step(src,direction))
+				if(C.d1 == turn(direction, 180) || C.d2 == turn(direction, 180))
+					connection_dirs += direction
+					break
+
+		for(var/direction in connection_dirs)
+			var/obj/structure/cable/C = new/obj/structure/cable(src.loc)
+			C.d1 = 0
+			C.d2 = direction
+			C.icon_state = "[C.d1]-[C.d2]"
+			C.breaker_box = src
+
+			var/datum/powernet/PN = new()
+			PN.add_cable(C)
+
+			C.mergeConnectedNetworks(C.d2)
+			C.mergeConnectedNetworksOnTurf()
+
+			if(C.d2 & (C.d2 - 1))// if the cable is layed diagonally, check the others 2 possible directions
+				C.mergeDiagonalsNetworks(C.d2)
+	else
+		icon_state = icon_state_off
+		for(var/obj/structure/cable/C in src.loc)
+			qdel(C)
```

Comment:
that proc doesn't remove the wire itself, you have to make sure it wont automatically recreate the connection later